adobe/spectrum-web-components
# feat: screen reader support for Calendar and DateTimePicker

## Description

This PR aims to offer screen reader support for the most common use-cases of these components.

### The calendar:
- Announces when a date is selected
- Announces when the month is changed
- Announces when the user navigates using Arrows or VO + Arrows
- Announces the month and year when going into the component
- Announces the type of action on previous/next month buttons
- New `labels` property to override english-default labels

### The DateTimePicker:
- Announces the current focused segment as a stepper with the correct type translated according to the locale ('year'/'month' etc.)
- Announces if the current focused segment is Empty
- Announces the new value of the segment when it gets changed 
- Announces the type of button 'Calendar'
- Announces the spelled out date value (or 'Empty') when going into the component
- New `labels` property to override english-default labels

## Related issue(s)

- https://github.com/adobe/spectrum-web-components/issues/2559
- https://github.com/adobe/spectrum-web-components/issues/2560

## Motivation and context

Screen reader is one of the basic testings - the minimum level expected for any product of Adobe. 

## How has this been tested?

mainly Chrome with macOS VoiceOver (cmd + F5)

    1. Go to the Calendar storybook 
    2. Tab to the Calendar component
    3. Observe that the month and year get announced
    4. Tab to the days grid and use VO + Arrows to navigate through the dates
    5. Observe that the VoiceOver announces the full dates for each day in the grid
    6. Select a date using VO + space
    7. Observe that the new selected value is announced
    1. Go to the Calendar storybook
    2. Tab to the next/previous month buttons
    3. Observe that the action is announced ('Previous'/'Next')
    4. Change the month using VO + space
    5. Observe that the new month and year get announced
    1. Go to the DateTimePicker storybook
    2. Tab to the DateTimePicker component
    3. Observe that the value gets announced (or 'Empty')
    4. Tab through the segments (year/month/hour etc.)
    6. Observe that the type of the segment gets announced along with the value
    7. Change the value of the segment using Arrow Up/ Arrow Down or Type In
    8. Observe that the new value gets announced



## Screenshots (if appropriate)

## Types of changes



-   [ x New feature (non-breaking change which adds functionality)

## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
